git/githistory/rewriter.go

```diff
@@ -172,6 +175,33 @@ func NewRewriter(db *odb.ObjectDatabase, opts ...rewriterOption) *Rewriter {
 	return rewriter
 }
 
+// ScanForPointers scans through the range of commits given by
```

Comment:
Nice. Historically, we've used package `lfs`'s `GitScanner` type to do such a thing. It's backed by the now-outdated `git/git.go`, which shells out to `git(1)`, but it does roughly what you're doing here. I'd like someday to replace it with a `git/odb`-backed implementation, but for now that doesn't exist. Instead of growing two separate implementations, how about we lean on what's already provided and replace it over time in a later PR?

```diff
diff --git a/commands/command_migrate_export.go b/commands/command_migrate_export.go
index b234eac9..53e64f49 100644
--- a/commands/command_migrate_export.go
+++ b/commands/command_migrate_export.go
@@ -104,10 +104,15 @@ func migrateExportCommand(cmd *cobra.Command, args []string) {
 	// If we have a valid remote, pre-download all objects using the Transfer Queue
 	if remoteURL := getAPIClient().Endpoints.RemoteEndpoint("download", cfg.Remote()).Url; remoteURL != "" {
 		q := newDownloadQueue(getTransferManifestOperationRemote("Download", cfg.Remote()), cfg.Remote())
-		if err := rewriter.ScanForPointers(q, opts, gitfilter); err != nil {
+		scan := lfs.NewGitScanner(func(p *lfs.WrappedPointer, err error) {
+			if err != nil {
+				return
+			}
+			q.Add(p.Name, p.Name, p.Oid, p.Size)
+		})
+		if err := scan.ScanRefComplex(opts.Include, opts.Exclude, nil); err != nil {
 			ExitWithError(err)
 		}
-
 		q.Wait()
 
 		for _, err := range q.Errors() {
diff --git a/lfs/gitscanner.go b/lfs/gitscanner.go
index 15a6d58c..20d6e49d 100644
--- a/lfs/gitscanner.go
+++ b/lfs/gitscanner.go
@@ -92,6 +92,17 @@ func (s *GitScanner) ScanLeftToRemote(left string, cb GitScannerFoundPointer) er
 	return scanRefsToChan(s, callback, left, "", s.opts(ScanLeftToRemoteMode))
 }
 
+func (s *GitScanner) ScanRefComplex(include, exclude []string, cb GitScannerFoundPointer) error {
+	callback, err := firstGitScannerCallback(cb, s.FoundPointer)
+	if err != nil {
+		return err
+	}
+
+	opts := s.opts(ScanRefsMode)
+	opts.SkipDeletedBlobs = false
+	return scanRefsToChan(s, callback, include, exclude, opts)
+}
+
 // ScanRefRange scans through all commits from the given left and right refs,
 // including git objects that have been modified or deleted.
 func (s *GitScanner) ScanRefRange(left, right string, cb GitScannerFoundPointer) error {
@@ -102,7 +113,7 @@ func (s *GitScanner) ScanRefRange(left, right string, cb GitScannerFoundPointer)
 
 	opts := s.opts(ScanRefsMode)
 	opts.SkipDeletedBlobs = false
-	return scanRefsToChan(s, callback, left, right, opts)
+	return scanLeftRightToChan(s, callback, left, right, opts)
 }
 
 // ScanRefWithDeleted scans through all objects in the given ref, including
diff --git a/lfs/gitscanner_refs.go b/lfs/gitscanner_refs.go
index e66daee2..024ff117 100644
--- a/lfs/gitscanner_refs.go
+++ b/lfs/gitscanner_refs.go
@@ -33,10 +33,7 @@ func (s *lockableNameSet) Check(blobSha string) (string, bool) {
 
 func noopFoundLockable(name string) {}
 
-// scanRefsToChan takes a ref and returns a channel of WrappedPointer objects
-// for all Git LFS pointers it finds for that ref.
-// Reports unique oids once only, not multiple times if >1 file uses the same content
-func scanRefsToChan(scanner *GitScanner, pointerCb GitScannerFoundPointer, refLeft, refRight string, opt *ScanRefsOptions) error {
+func scanRefsToChan(scanner *GitScanner, pointerCb GitScannerFoundPointer, include, exclude []string, opt *ScanRefsOptions) error {
 	if opt == nil {
 		panic("no scan ref options")
 	}
@@ -134,3 +131,10 @@ func revListShas(include, exclude []string, opt *ScanRefsOptions) (*StringChanne
 
 	return NewStringChannelWrapper(revs, errs), nil
 }
+
+// scanLeftRightToChan takes a ref and returns a channel of WrappedPointer objects
+// for all Git LFS pointers it finds for that ref.
+// Reports unique oids once only, not multiple times if >1 file uses the same content
+func scanLeftRightToChan(scanner *GitScanner, pointerCb GitScannerFoundPointer, refLeft, refRight string, opt *ScanRefsOptions) error {
+	return scanRefsToChan(scanner, pointerCb, []string{refLeft}, []string{refRight}, opt)
+}
```

Reply:
Thanks! I agree with you, we should be using the `GitScanner` for this code - I'll go ahead and replace what I've done with the `ScanForPointers` function